cmd/handlers/handler.go

```diff
@@ -1,22 +1,60 @@
 package handlers
 
 import (
+	"encoding/json"
+	"fmt"
 	"github.com/go-chi/chi/v5"
 	"github.com/kholodmv/go-service/cmd/storage"
+	"github.com/kholodmv/go-service/internal/gzip"
+	"github.com/kholodmv/go-service/internal/logger"
+	"github.com/kholodmv/go-service/internal/models"
+	"os"
 )
 
 type Handler struct {
+	router     chi.Router
 	repository storage.MetricRepository
 }
 
-func NewHandler(repository storage.MetricRepository) *Handler {
-	return &Handler{
+func NewHandler(router chi.Router, repository storage.MetricRepository, filename string, restore bool) *Handler {
+	h := &Handler{
 		repository: repository,
+		router:     router,
 	}
+
+	if restore {
+		file, err := os.OpenFile(filename, os.O_RDONLY|os.O_CREATE, 0666)
+		if err != nil {
+			fmt.Printf("Сan not open file: %s\n", err)
+		}
+		defer file.Close()
+
+		metrics := &[]models.Metrics{}
```

Comment:
лишнее, внутри и так лежит ссылка на массив
```suggestion
		metrics := []models.Metrics{}
```